test/models/item_test.rb

```diff
@@ -28,11 +36,11 @@ class ItemTest < ActiveSupport::TestCase
     second_item.mark
 
     second_item.reload
-    assert_equal second_item.list.incompleted_items.count, second_item.position
+    assert_equal second_item.list.items.where(done: false).count, second_item.position
   end
 
   test "should move newly created item to the bottom" do
     item = FactoryGirl.create(:with_valid_attributes)
-    assert_equal item.list.incompleted_items.count, item.position
+    assert_equal item.list.items.where(done: false).count, item.position
```

Comment:
@vlasiak, I suggest to extract a scope for `where(done: false)`, and reuse it in `Item#set_highest_position` method. 
Also, you can try `not` method of ActiveRecord.